### DISCUSSION TITLE
What is the difference between the normal and database container?

### DISCUSSION THREAD

**Discussion by freinold** [BerriAI/litellm#5276] [Q&A]
When comparing the `Dockerfile` to the `Dockerfile.database`, the only difference to me seems the order of install instructions. (screenshot added below)

![7c27a34f-781c-40ec-8d9a-ddadafedeaa8](https://github.com/user-attachments/assets/bd7300c8-8837-47ae-9f84-fbd4e099b02b)

How the the database image different and reduces build time, [like advertised in the documentation?](https://docs.litellm.ai/docs/proxy/deploy#docker-kubernetes-helm-chart)

**Comment by mkagit** [comment]
Also about between `ghcr.io/berriai/litellm:main-latest` and `ghcr.io/berriai/litellm-database:main-latest` it is not clear the difference

**Reply by ovizii** [reply]
I can confirm, I have read the docs but still can't figure out the difference. Also, the instructions (deployment options) found here: https://docs.litellm.ai/docs/proxy/deploy#deployment-options show 4 different options, but I can't find the difference between option 2 and 4.

![image](https://github.com/user-attachments/assets/7c7f01f3-ae7d-4e42-8cdf-f1fdc6ab15e3)

Option 2 links to instructions which mention this variable `DATABASE_URL=postgresql://<user>:<password>@<host>:<port>/<dbname>` and is using this image: `ghcr.io/berriai/litellm-database:main-latest `

Option 4 does not link anywhere, or link is broken.

**Reply by ovizii** [reply]
I just wanted to add that the docs mention to 

> Deploy with Redis: Use Redis when you need litellm to load balance across multiple litellm containers 

but the config: https://docs.litellm.ai/docs/proxy/config_settings 
says that Redis is also used for caching:
```
  # Caching settings
  cache: true 
  cache_params:        # set cache params for redis
    type: redis        # type of cache to initialize

```

**Comment by ghost** [comment]
Hey @freinold this is a good question. When we initially started publishing docker images, we had a separate one (litellm-database) where we built the prisma binary. 

I believe over time, this has also been added to the basic litellm image. We can probably consolidate to just the litellm image
